File: core/src/main/java/me/dm7/barcodescanner/core/CameraWrapper.java
package me.dm7.barcodescanner.core;

import android.hardware.Camera;
import android.support.annotation.NonNull;

public class CameraWrapper {
    public final Camera camera;
    public final int cameraId;

    private CameraWrapper(@NonNull Camera camera, int cameraId) {
        if (camera == null) {
            throw new NullPointerException("Camera cannot be null");
        }
        this.camera = camera;
        this.cameraId = cameraId;
    }

    public static CameraWrapper getWrapper(Camera camera, int cameraId) {
        if (camera == null) {
            return null;
        } else {
            return new CameraWrapper(camera, cameraId);
        }
    }
}

File: core/src/main/java/me/dm7/barcodescanner/core/CameraWrapper.kt
package me.dm7.barcodescanner.core

import android.hardware.Camera

class CameraWrapper private constructor(var mCamera: Camera?, val mCameraId: Int) {

    init {
        if (mCamera == null) {
            throw NullPointerException("Camera cannot be null")
        }
    }

    companion object {

        fun getWrapper(camera: Camera?, cameraId: Int): CameraWrapper? {
            return if (camera == null) {
                null
            } else {
                CameraWrapper(camera, cameraId)
            }
        }
    }
}